src/deluge/modulation/arpeggiator.cpp

```diff
@@ -367,6 +367,14 @@ bool ArpeggiatorBase::evaluateRhythm(bool isRatchet) {
 	return arpRhythmPatterns[rhythm].steps[rhythmPatternIndex];
 }
 
+// Returns if the arpeggiator should play the next note or not
+bool ArpeggiatorBase::evaluateNoteProbability(bool isRatchet) {
+	// If it is a rachet, use the last value, but it it is not a ratchet, calculate a new probability
+	int32_t randomChance = random(65535);
```

Comment:
@m-m-adams should we be using getNoise here for performance? (or even `getNoise() >> 16`)?

Reply:
Since this PR is from long time ago, which I left abandoned due to a bug that I was just able to resolve this week after looking with different eyes, I have not a fresh memory of it, but I recall having used getNoise() and not getting truly random values. random() was the only way I got to get *real* probability

Reply:
random is being used also inside  void ArpeggiatorBase::maybeSetupNewRatchet(ArpeggiatorSettings* settings)

Reply:
We probably just need to improve our local pseudo-random algo at some point then tbh. oh well 😅